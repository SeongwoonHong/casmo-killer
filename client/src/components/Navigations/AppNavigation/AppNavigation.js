import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import classnames from 'classnames';

import './AppNavigation.scss';

import { MainMenuRoutes } from '../../../routers';

import MainMenu from './MainMenu';
import UserMenu from './UserMenu';
import SubMenu from './SubMenu';

class AppNavigation extends Component {

  componentDidUpdate(prevProps) {

    const {
      location, layout, closeAllMenu
    } = this.props;

    if (location !== prevProps.location && layout.breakPoint !== 'xl') {
      closeAllMenu();
    }

  }

  render() {

    const {
      location, user, layout, closeAllMenu, toggleMenu, toggleUserMenu, logout
    } = this.props;

    const SubMenus = (routes) => {
      return routes.map((route) => {
        if (route.children) {
          return (
            <Route
              exact={ route.exact }
              key={ route.name || 'root' }
              path={ route.path }
              render={props => (
                <SubMenu
                  { ...props }
                  parent={ route.path }
                  items={ route.children } />
              ) }
            />
          );
        }
        return null;
      });
    };

    return ([
      <div
        key={ 0 }
        role="button"
        tabIndex={ 0 }
        className={ classnames('App-navigation__overlay', {
          active: layout.isMainMenuVisible || layout.isUserMenuVisible
        })}
        onClick={ closeAllMenu }
        onKeyDown={ () => {} }
      />,

      <nav
        key={ 1 }
        className="App-navigation">
        <div className="App-navigation__row component-row">
          <MainMenu
            active={ layout.isMainMenuVisible }
            menus={ MainMenuRoutes }
            toggleMenu={ toggleMenu } />
          <UserMenu
            active={ layout.isUserMenuVisible }
            user={ user }
            layout={ layout }
            currentRoute={ location.pathname }
            toggleUserMenu={ toggleUserMenu }
            logout={ logout } />
        </div>
        <Switch>
          { SubMenus(MainMenuRoutes) }
        </Switch>
      </nav>
    ]);

  }

}

export default AppNavigation;
