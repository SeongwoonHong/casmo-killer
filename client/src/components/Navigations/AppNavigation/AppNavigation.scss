.App-navigation {
    &__overlay {
        @include abs-elem(0, 0, 0, 0, absolute);
        background-color: rgba(0, 0, 0, .7);
        display: none;
        z-index: 88;
        &.active {
            display: block;
            @media screen and (min-width: 992px) {
                display: none;
            }
        }
    }
    @media screen and (min-width: 992px) {
        background-color: palette(Grey, 800);
        border-bottom: 2px solid palette(Grey, 900);
        position: sticky;
        top: 0;
        z-index: 99;
        &__row {
            @include flex-box(space-between);
        }
    }
}
