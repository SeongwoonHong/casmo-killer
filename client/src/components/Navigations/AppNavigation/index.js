import { connect } from 'react-redux';
import * as actions from '@actions';

import AppNavigation from './AppNavigation';

const mapStateToProps = (state) => {
  return {
    user: state.user.user,
    layout: state.layout
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    closeAllMenu: () => {
      dispatch(actions.toggleMenu(false));
      dispatch(actions.toggleUserMenu(false));
    },
    toggleMenu: (payload = null) => dispatch(actions.toggleMenu(payload)),
    toggleUserMenu: payload => dispatch(actions.toggleUserMenu(payload)),
    logout: () => dispatch(actions.logout())
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(AppNavigation);
