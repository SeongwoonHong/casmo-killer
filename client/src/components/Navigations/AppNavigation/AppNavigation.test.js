/* eslint react/jsx-boolean-value: 0 */
import React from 'react';
import { shallow } from 'enzyme';

import AppNavigation from './AppNavigation';

describe('<AppNavigation />', () => {

  let component = null;

  beforeEach(() => {
    component = shallow(
      <AppNavigation
        layout={ {
          isMainMenuVisible: true,
          isUserMenuVisible: true
        }}
        location={ {
          pathname: '/'
        }} />
    );
  });

  it('renders the component consistently', () => {

    expect(component).toMatchSnapshot();

  });

});
