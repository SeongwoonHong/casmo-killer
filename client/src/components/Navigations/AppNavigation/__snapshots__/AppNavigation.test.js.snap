// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppNavigation /> renders the component consistently 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppNavigation
    layout={
        Object {
            "isMainMenuVisible": true,
            "isUserMenuVisible": true,
          }
    }
    location={
        Object {
            "pathname": "/",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <div
      className="App-navigation__overlay active"
      onClick={undefined}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
/>,
    <nav
      className="App-navigation"
>
      <div
            className="App-navigation__row component-row"
      >
            <MainMenu
                  active={true}
                  menus={
                        Array [
                              Object {
                                "exact": true,
                                "path": "/",
                              },
                              Object {
                                "children": Array [
                                  Object {
                                    "name": "자유게시판",
                                    "path": "/free",
                                  },
                                  Object {
                                    "name": "나의 커뮤니티",
                                    "path": "/myCommunity",
                                  },
                                  Object {
                                    "name": "즐겨찾기",
                                    "path": "/favourites",
                                  },
                                  Object {
                                    "name": "커뮤니티 All",
                                    "path": "/communityAll",
                                  },
                                ],
                                "icon": "group",
                                "main": [Function],
                                "name": "커뮤니티",
                                "path": "/community",
                              },
                              Object {
                                "children": Array [
                                  Object {
                                    "name": "오늘의 맛집",
                                    "path": "/trending",
                                  },
                                  Object {
                                    "name": "랭킹",
                                    "path": "/rank",
                                  },
                                ],
                                "icon": "free_breakfast",
                                "name": "맛집",
                                "path": "/restaurants",
                              },
                              Object {
                                "children": Array [
                                  Object {
                                    "name": "최신질문",
                                    "path": "/recent",
                                  },
                                  Object {
                                    "name": "내 질문",
                                    "path": "/myquestions",
                                  },
                                  Object {
                                    "name": "내 답변",
                                    "path": "/myanswers",
                                  },
                                ],
                                "icon": "forum",
                                "name": "Q&A",
                                "path": "/qna",
                              },
                              Object {
                                "children": Array [
                                  Object {
                                    "name": "인기상승",
                                    "path": "/whatever",
                                  },
                                  Object {
                                    "name": "아무거나",
                                    "path": "/whatevery",
                                  },
                                ],
                                "icon": "event",
                                "name": "정보",
                                "path": "/info",
                              },
                              Object {
                                "main": [Function],
                                "name": "게시글 목록",
                                "path": "/articles",
                              },
                              Object {
                                "main": [Function],
                                "name": "게시글",
                                "path": "/article",
                              },
                            ]
                  }
                  toggleMenu={undefined}
            />
            <OnClickOutside(UserMenu)
                  active={true}
                  currentRoute="/"
                  eventTypes={
                        Array [
                              "mousedown",
                              "touchstart",
                            ]
                  }
                  excludeScrollbar={false}
                  layout={
                        Object {
                              "isMainMenuVisible": true,
                              "isUserMenuVisible": true,
                            }
                  }
                  logout={undefined}
                  outsideClickIgnoreClass="ignore-react-onclickoutside"
                  preventDefault={false}
                  stopPropagation={false}
                  toggleUserMenu={undefined}
                  user={undefined}
            />
      </div>
      <Switch>
            <Route
                  exact={undefined}
                  path="/community"
                  render={[Function]}
            />
            <Route
                  exact={undefined}
                  path="/restaurants"
                  render={[Function]}
            />
            <Route
                  exact={undefined}
                  path="/qna"
                  render={[Function]}
            />
            <Route
                  exact={undefined}
                  path="/info"
                  render={[Function]}
            />
      </Switch>
</nav>,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <div
        className="App-navigation__overlay active"
        onClick={undefined}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
/>,
      <nav
        className="App-navigation"
>
        <div
                className="App-navigation__row component-row"
        >
                <MainMenu
                        active={true}
                        menus={
                                Array [
                                        Object {
                                          "exact": true,
                                          "path": "/",
                                        },
                                        Object {
                                          "children": Array [
                                            Object {
                                              "name": "자유게시판",
                                              "path": "/free",
                                            },
                                            Object {
                                              "name": "나의 커뮤니티",
                                              "path": "/myCommunity",
                                            },
                                            Object {
                                              "name": "즐겨찾기",
                                              "path": "/favourites",
                                            },
                                            Object {
                                              "name": "커뮤니티 All",
                                              "path": "/communityAll",
                                            },
                                          ],
                                          "icon": "group",
                                          "main": [Function],
                                          "name": "커뮤니티",
                                          "path": "/community",
                                        },
                                        Object {
                                          "children": Array [
                                            Object {
                                              "name": "오늘의 맛집",
                                              "path": "/trending",
                                            },
                                            Object {
                                              "name": "랭킹",
                                              "path": "/rank",
                                            },
                                          ],
                                          "icon": "free_breakfast",
                                          "name": "맛집",
                                          "path": "/restaurants",
                                        },
                                        Object {
                                          "children": Array [
                                            Object {
                                              "name": "최신질문",
                                              "path": "/recent",
                                            },
                                            Object {
                                              "name": "내 질문",
                                              "path": "/myquestions",
                                            },
                                            Object {
                                              "name": "내 답변",
                                              "path": "/myanswers",
                                            },
                                          ],
                                          "icon": "forum",
                                          "name": "Q&A",
                                          "path": "/qna",
                                        },
                                        Object {
                                          "children": Array [
                                            Object {
                                              "name": "인기상승",
                                              "path": "/whatever",
                                            },
                                            Object {
                                              "name": "아무거나",
                                              "path": "/whatevery",
                                            },
                                          ],
                                          "icon": "event",
                                          "name": "정보",
                                          "path": "/info",
                                        },
                                        Object {
                                          "main": [Function],
                                          "name": "게시글 목록",
                                          "path": "/articles",
                                        },
                                        Object {
                                          "main": [Function],
                                          "name": "게시글",
                                          "path": "/article",
                                        },
                                      ]
                        }
                        toggleMenu={undefined}
                />
                <OnClickOutside(UserMenu)
                        active={true}
                        currentRoute="/"
                        eventTypes={
                                Array [
                                        "mousedown",
                                        "touchstart",
                                      ]
                        }
                        excludeScrollbar={false}
                        layout={
                                Object {
                                        "isMainMenuVisible": true,
                                        "isUserMenuVisible": true,
                                      }
                        }
                        logout={undefined}
                        outsideClickIgnoreClass="ignore-react-onclickoutside"
                        preventDefault={false}
                        stopPropagation={false}
                        toggleUserMenu={undefined}
                        user={undefined}
                />
        </div>
        <Switch>
                <Route
                        exact={undefined}
                        path="/community"
                        render={[Function]}
                />
                <Route
                        exact={undefined}
                        path="/restaurants"
                        render={[Function]}
                />
                <Route
                        exact={undefined}
                        path="/qna"
                        render={[Function]}
                />
                <Route
                        exact={undefined}
                        path="/info"
                        render={[Function]}
                />
        </Switch>
</nav>,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
